File: features/lender/src/main/java/com/nivasafinance/features/lender/lender/exception/LenderNotFoundException.java
package com.nivasafinance.features.lender.lender.exception;

import com.nivasafinance.common.exception.ExceptionUtils;
import com.nivasafinance.common.exception.ResourceNotFoundException;
import org.springframework.context.MessageSource;

import java.io.Serial;
import java.util.UUID;

public class LenderNotFoundException extends ResourceNotFoundException {

    @Serial
    private static final long serialVersionUID = 1L;

    public LenderNotFoundException(UUID lenderId, MessageSource messageSource) {
        super(ExceptionUtils.createLocalizedMessage(
                "error.lender.not.found",
                new Object[]{lenderId.toString()},
                messageSource
        ));
    }
}

File: features/lender/src/main/java/com/nivasafinance/features/lender/lender/exception/LenderNotFoundException.kt
package com.nivasafinance.features.lender.lender.exception

import com.nivasafinance.common.exception.ExceptionUtils
import com.nivasafinance.common.exception.ResourceNotFoundException
import org.springframework.context.MessageSource
import java.util.UUID

class LenderNotFoundException(
    lenderId: UUID,
    messageSource: MessageSource
) : ResourceNotFoundException(
    ExceptionUtils.createLocalizedMessage(
        "error.lender.not.found",
        arrayOf(lenderId.toString()),
        messageSource
    )
)

class LenderKeyNotFoundException(
    key: String,
    messageSource: MessageSource
) : ResourceNotFoundException(
    ExceptionUtils.createLocalizedMessage(
        "error.lender.key.not.found",
        arrayOf(key),
        messageSource
    )
)
